Classes/Localization/LocalizationFactory.php

```diff
@@ -30,9 +30,12 @@ class LocalizationFactory extends \TYPO3\CMS\Core\Localization\LocalizationFacto
 	 */
 	public function getParsedData($fileReference, $languageKey, $charset = '', $errorMode = 0, $isLocalizationOverride = FALSE) {
 		$data = parent::getParsedData($fileReference, $languageKey, $charset, $errorMode, $isLocalizationOverride);
-		if (FALSE === $this->isWhitelisted($fileReference) || TRUE === $this->isBlacklisted($fileReference)) {
+		if ((FALSE === $this->isWhitelisted($fileReference) || TRUE === $this->isBlacklisted($fileReference))) {
 			return $data;
 		}
+		if (TRUE === empty($GLOBALS['TSFE']->lang)) {
+			$GLOBALS['TSFE']->lang = $languageKey;
```

Comment:
We definitely do not want to be doing this. It is far from the job of FLLL to be manipulating the frontend context - we should instead plan and simple avoid processing when no language is set and no default is given.

Be aware also that the "TSFE" object has changed in recent TYPO3 versions. The public property may no longer be trustworthy or the correct way of resolving the current language.